integration/src/main/java/scripts/testscripts/DatasetOperations.java

```diff
@@ -161,12 +164,26 @@ private void previewUserJourney(StorageSystem storageSystem, String sourceId)
     datasetId = null;
   }
 
+  private ObjectNode expectedMetadataResponse(ObjectNode metadata, StorageSystem storageSystem) {
+    ObjectNode expectedMetadata = metadata.deepCopy().put("id", datasetId.toString());
```

Comment:
Rather than using `deepCopy()` I think it'd be better to move the metadata out of a static and into a regular field. Having mutable objects like `ObjectNode` as static members isn't great in general IMO and we don't want test writers to have to think about when they might need to copy them when writing a test.